### PR TITLE
Update for 3Dconnexion NavLib integration

### PR DESCRIPTION
**Pull request by pskowronskiTDx**
Hello,

The following changes address some issues with the new integration, which were recently found:

- Fixed an crash occuring if QuickZoom is used on geometry clipped by the view volume,
- Fixed a bug causing an center of rotation stop updating it's position in ortho view,
- Fixed QuickZoom for high DPI screens (support for variable scaling)

### CODE REVIEW

**Comment by maxwxyz** [comment]
@pskowronskiTDx can this safely be backported to 1.0?

**Comment by pskowronskiTDx** [comment]
@maxwxyz Yes, I do not expect any issues with that.

File: src/Gui/3Dconnexion/navlib/NavlibPivot.cpp [hunk 1]
```diff
@@ -219,8 +245,24 @@ long NavlibInterface::SetHitDirection(const navlib::vector_t& direction)
...
+        auto pCamera = getCamera<SoCamera*>();
+        if (pCamera == nullptr) {
+            return navlib::make_result_code(navlib::navlib_errc::no_data_available);
+        }
+
+        SbVec3f position = pCamera->position.getValue();
+        ray.origin = position + orthoNearDistance * ray.direction;
+
+        return 0;
+    }
```
Comment (hyarion) [inline src/Gui/3Dconnexion/navlib/NavlibPivot.cpp:265]: This looks like a bug. If `isPerspective` is `true` then there's no defined return value.
Shouldn't the `return 0` be placed outside the else block?
```suggestion
    }
    return 0;
```

File: src/Gui/3Dconnexion/navlib/NavlibNavigation.cpp [hunk 2]
```diff
@@ -101,11 +101,17 @@ long NavlibInterface::GetPointerPosition(navlib::point_t& position) const
...
         QPoint viewPoint = currentView.pView3d->mapFromGlobal(QCursor::pos());
         viewPoint.setY(currentView.pView3d->height() - viewPoint.y());
+
+        double scaling = inventorViewer->devicePixelRatio();
+        viewPoint *= scaling;
+
         SbVec3f worldPosition =
             inventorViewer->getPointOnFocalPlane(SbVec2s(viewPoint.x(), viewPoint.y()));
-
+
```
Comment (hyarion) [inline src/Gui/3Dconnexion/navlib/NavlibNavigation.cpp:110]: Remove trailing whitespace
```suggestion

```

**Review by hyarion — CHANGES_REQUESTED** [review]

File: src/Gui/3Dconnexion/navlib/NavlibPivot.cpp (hunk 1 above)
Comment (pskowronskiTDx) [reply src/Gui/3Dconnexion/navlib/NavlibPivot.cpp:265]: Yes, you are right. I have overlooked this.

**Comment by adrianinsaval** [comment]
backported

**Pull request merged**